src/core/mac/mac.hpp

```diff
@@ -341,6 +341,14 @@ class Mac : public InstanceLocator, public SubMac::Callbacks
      */
     void SetSupportedChannelMask(const ChannelMask &aMask);
 
+    /**
+     * This method returns the phy supported channel mask.
+     *
+     * @returns The phy supported channel mask.
+     *
+     */
+    uint32_t GetPhySupportedChannelMask(void) const;
```

Comment:
Suggestion: How about making this new method a private one and allowing others to use `GetSupportedChannelMask()`. This way, the supported channel mask would be configurable by both radio platform and user (through the public API `otLinkSetSupportedChannelMask`).

Reply:
We have another otLinkGetPhySupportedChannelMask() will call it, so keep it as public. Generally, the SupportedChannelMask and PhySupportedChannelMask could be different. User can't change the PhySupportedChannelMask, but can change the SupportedChannelMask.

Reply:
I am thinking that we don't expose `otLinkGetPhySupportedChannelMask()` as a public API.

Basically, from OT perspective, we allow supported channel mask to be specified from higher layer (through public API `otLinkSetSupportedChannelMask`) and/or from radio layer (through the new `otPlatRadioGetSupportedChannelMask()` API).

In `Mac`  we keep track of the mask which is the combination/intersection of the two (from public API and radio API) and everywhere else in OT core we use the `Mac::GetSupportedChannelMask()`.  

The public API `otLinkGetSupportedChannelMask()` would also get the mask from `Mac` (which it already does) so it would be always the mask that is being used by OT core.

Thoughts?

Reply:
Agree, I moved the `otLinkGetPhySupportedChannelMask()` (as well as two other otLinkGetPhy*()) to be radio platform API, and have a weak implementation for them. Then in this case, `Mac::GetPhySupportedChannelMask()` is not necessary more, removed it.